libs/device/feature/src/lib/data-transfer/data-transfer.service.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { Result, ResultObject } from "Core/core/builder"
+import { IpcEvent } from "Core/core/decorators"
+import { AppError } from "Core/core/errors"
+import { DeviceManager } from "Core/device-manager/services"
+import { DeviceId } from "Core/device/constants/device-id"
+import { ApiResponse } from "Core/device/types/mudita-os"
+import {
+  APIDataTransferServiceEvents,
+  DataTransferDomain,
+  GeneralError,
+  PreDataTransfer,
+} from "device/models"
+import random from "lodash/random"
+
+export class APIDataTransferService {
+  constructor(private deviceManager: DeviceManager) {}
+
+  @IpcEvent(APIDataTransferServiceEvents.StartPreDataTransfer)
+  public async startPreDataTransfer({
+    domains,
+    deviceId,
+  }: {
+    domains: DataTransferDomain[]
+    deviceId?: DeviceId
+  }): Promise<ResultObject<PreDataTransfer>> {
+    const device = deviceId
+      ? this.deviceManager.getAPIDeviceById(deviceId)
+      : this.deviceManager.apiDevice
+
+    if (!device) {
+      return Result.failed(new AppError(GeneralError.NoDevice, ""))
+    }
```

Comment:
⚪ (A question not necessarily related to the requested changes for this pull request)

Can this logic not be part of a lower layer? (for example DeviceManager) I've noticed that we duplicate this code in almost every service.